### ISSUE TITLE
Goto in lhs of match arm for simple identifier doesn't work

### ISSUE THREAD

**Issue by edwin0cheng** [rust-analyzer/rust-analyzer#2606]
Minimun test case:

```rust
enum Enum {A}
use Enum::*;

fn main() {
    match Enum::A {
        <|>A => {}    // Doesn't work
    }

    let a = <|>A;  // It works
}
``

**Comment by matklad** [comment]
This is long-standing bug about distinguishing `match Enum::A { A => (/*this is a const*/), a => (/* this is a binding*/)}`

**Comment by edwin0cheng** [comment]
It is fixed by #3262, thanks
